uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -80,6 +96,33 @@ define([
       return activeTab.fields;
     }
 
+    /**
+     * Indexes fields for quicker access and sets them to the component
+     */
+    function indexFields () {
+      vm.sections.forEach(function (section) {
+        section.tabs.forEach(function (tab) {
+          tab.fields.forEach(function (field) {
+            vm.fieldsIndexed[field.name] = field;
+          });
+        });
+      });
```

Comment:
Prepare yourself to be blown away:

```js
vm.fieldsIndexed = _.chain(vm.sections).flatMap('tabs').flatMap('fields').keyBy('name').value();
```

Reply:
Damn @reneolivo 

![image](https://user-images.githubusercontent.com/3973243/47032222-bc8e1700-d169-11e8-80b5-be7ba5063601.png)